be/src/storage/lake/txn_log_applier.cpp

```diff
@@ -126,11 +126,30 @@ class PrimaryKeyTxnLogApplier : public TxnLogApplier {
         _index_entry = nullptr;
     }
 
-    Status apply(const TxnLogPB& log) override {
+    Status check_rebuild_index(bool rebuild_pindex) {
+        if (!_has_rebuild_pindex && rebuild_pindex) {
+            for (const auto& sstable : _metadata->sstable_meta().sstables()) {
+                FileMetaPB file_meta;
+                file_meta.set_name(sstable.filename());
+                file_meta.set_size(sstable.filesize());
+                _metadata->mutable_orphan_files()->Add(std::move(file_meta));
+            }
+            _metadata->clear_sstable_meta();
+            _guard.reset(nullptr);
+            _index_entry = nullptr;
+            ASSIGN_OR_RETURN(_index_entry, _tablet.update_mgr()->rebuild_primary_index(
+                                                   _metadata, &_builder, _base_version, _new_version, _guard));
+            _has_rebuild_pindex = true;
+        }
+        return Status::OK();
+    }
+
+    Status apply(const TxnLogPB& log, bool rebuild_pindex) override {
```

Comment:
> Because `apply` is also used by Non-PK table, I think we can keep use `Status apply(const TxnLogPB& log)`, instead we pass `rebuild_pindex` to applier in constructor function. like:
> 
> ```
> PrimaryKeyTxnLogApplier(const Tablet& tablet, MutableTabletMetadataPtr metadata, int64_t new_version, bool rebuild_pindex)
>             : _tablet(tablet),
>               _metadata(std::move(metadata)),
>               _base_version(_metadata->version()),
>               _new_version(new_version),
>               _builder(_tablet, _metadata),
>               _need_rebuild_pindex(rebuild_pindex) {
>         _metadata->set_version(_new_version);
>     }
> ```
> 
> And we can add `_need_rebuild_pindex` to `PrimaryKeyTxnLogApplier`, after rebuild finish, `_need_rebuild_pindex` will be reset to false.

done